### ISSUE TITLE
Extend security reporting for LTS lines beyond their lifetimes

### ISSUE THREAD

**Issue by bengl** [nodejs/security-wg#1025]
Today, the reported (and fixed) security vulnerabilities in Node.js core are stored in this repo in JSON, but they only end up in the data if the vulnerability is fixed. This means that EOL release lines may or may not be vulnerable to any given vulnerability, if basing this on the data in this repo alone. AFAICT, vulnerabilities simply aren't reported on EOL lines. While this shouldn't be an issue, since folks are asked _not_ to run EOL versions of Node.js in production, in practice folks run EOL versions of Node.js long past their lifetimes.

My suggestion is that the security process be amended to report whether LTS Node.js release lines that have been EOL for _less than two years_ are vulnerable to any given vulnerability. I'm _not_ suggesting extending the LTS timeline in general (at least not in this issue 😄). I'm only suggesting that vulnerabilities be reported on them for an additional two years.

This would serve two purposes:

1. It would give users greater clarity into the scope of vulnerabilities, and whether or not they _need_ to upgrade _due to known vulnerabilities_ (yes, even though they should not run EOL versions anyway).
2. Publishing data on known vulnerabilities in release lines where they won't be fixed provides extra incentive to upgrade.

**Comment by tlhunter** [comment]
> and whether or not they need to upgrade due to known vulnerabilities

Or, another way of putting this: Such information would hopefully help convince folks running EOL versions of Node.js to upgrade.

**Comment by mcollina** [comment]
While I understand where you are coming from, assessing any vulnerability for old lines is tricky. In some cases we can't even build them on modern systems
or they won't run.

What we can change is set the minimum version to 2 LTS prior unless we know exactly when the vuln was introduced.

This does not change the workload,  and it something that can help updating.

**Comment by cjihrig** [comment]
I don't think we should do this. End of life should mean end of life. If we start looking at EOL + 2 years, people will start asking for EOL + 3 years and so on. In my experience, companies know that they are running an EOL version and will even go so far as hiring a company to provide support past EOL. I think we should allow for that model and not provide "official" post EOL support from a project of volunteers.

**Comment by Qard** [comment]
Can we test back to the x.0.0 of our oldest release and if the vulnerability is present _there_ communicate clearly that we've _tested_ back to there but _most likely_ the vulnerability exists _beyond_ that range? Like stating a _confirmed_ impacted range and suggesting something like `<=x.0.0` where `x.0.0` is the oldest version we tested against?

**Comment by mhdawson** [comment]
> It's hard to assess some H1 reports and if we expand the scope to EOL LTS versions that can be easily a nightmare for the ones in the rotation.

This is the key issue in my mind.  It requires more work, who is going to do that work?  Our support lifecycle is a balance between what the project takes on in terms of work and the needs of the community.  

I think that it would be great if people did this assessment and shared it. If there are a group of people who want to volunteer to do that work the project might help through some resources but I'd see is as an "unofficial" source without the project commiting to do the work.

**Comment by marco-ippolito** [comment]
> > It's hard to assess some H1 reports and if we expand the scope to EOL LTS versions that can be easily a nightmare for the ones in the rotation.
> 
> This is the key issue in my mind. It requires more work, who is going to do that work? Our support lifecycle is a balance between what the project takes on in terms of work and the needs of the community.
> 
> I think that it would be great if people did this assessment and shared it. If there are a group of people who want to volunteer to do that work the project might help through some resources (maybe similar model to unofficial builds) but I'd see is as an "unofficial" source without the project commiting to do the work.

I'd like to reopen this issue since I'd be interested in doing the work

**Comment by marco-ippolito** [comment]
@RafaelGSS it doesnt have to be retro-active. It can be done for new report. Whenever we triage a new report, I'll take care of reproducibility in EOL versions. This work will be sponsored by HeroDevs.

**Comment by RafaelGSS** [comment]
> @RafaelGSS it doesnt have to be retro-active. It can be done for new report. Whenever we triage a new report, I'll take care of reproducibility in EOL versions. This work will be sponsored by HeroDevs.

I'm not considering retro-active reports. My statement is valid for new reports too.

**Comment by marco-ippolito** [comment]
If we don't produce a CVE, users will think that even though their version is EOL, it is safe, which is absolutely not true. We still won't accept reports for older versions, but at least if reproducible, issue the CVE for that version.

**Comment by RafaelGSS** [comment]
For your second suggestion @mhdawson, see: #1401

**Comment by mhdawson** [comment]
@RafaelGSS thanks for the pointer, commented there.
